config/routes.rb

```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   devise_for :users
+  get '/i/:token', to: 'wishlists_for_guests#show'
```

Comment:
I don't if it may make some changes but try to use double quotation marks instead of single.